### DISCUSSION TITLE
GapCells=integer warhead

### DISCUSSION THREAD

**Discussion by Musketmarine** [Phobos-developers/Phobos#870] [Ideas and feedback]
`GapCells` when set to an integer above 0 will allow a warhead to inflict gap generator logic in the cells the warhead is detonated in. Uses Big Gap targeting flags. Defaults to 0 for all warheads.

Practical use would be a Gap Generator smokescreen or superweapon that shrouds an area.

```ini
[GapWH]
CellSpread=3
GapCells=3
BigGap.Affects=Enemy
```

Maybe GapCells could be a child of `BigGap` logic, so syntax would become `BigGap.GapCells=integer`.

**Comment by GraionDilach** [comment]
> syntax would become BigGap.GapCells=integer.

Sounds redundant to me. `BigGap.Cells` perhaps moreso.

**Comment by RexDraco** [comment]
For the sake of versatility, I suggest differentiating between global and weapon targeting, preferably in a way we can have both on the same weapon:

BigGap.AffectsImpact= (enemy, friend, ally, self) ; impact as in warhead. If you hit an enemy with this weapon and it is vulnerable, this happens. percentage influences BigGap.ImpactDuration
BigGap.AffectsGlobal= (enemy, friend, ally, self) ; impact as in global, like Phobos currently operates. This will be handy for smoke screens, but otherwise is like above which is handy for hacking. It is possible to have both, punishing the player it hits.

BigGap.ImpactDuration= (int) ; how long the impact variant will last, versus is taken into account
BigGap.GlobalDuration= (int) ; how long the global variant will last, versus does nothing

BigGap.ImpactLocation= (center, target) ; does it spawn the GapCells only at the center of the warhead or does it spawn multiple circles of gaps on every enemy the WH connects to? Meaning the bigger group of enemies, the more shroud created.
